addons/purchase/models/purchase.py

```diff
@@ -127,6 +126,85 @@ def _compute_invoice(self):
     company_id = fields.Many2one('res.company', 'Company', required=True, index=True, states=READONLY_STATES, default=lambda self: self.env.company.id)
     currency_rate = fields.Float("Currency Rate", compute='_compute_currency_rate', compute_sudo=True, store=True, readonly=True, help='Ratio between the purchase order currency and the company currency')
 
+    reminder_confirmed = fields.Boolean("RFQ Confirmed", default=False, readonly=True, help="True if the reminder email is confirmed by the vendor.")
+    po_confirmed = fields.Boolean("PO Confirmed", default=False, readonly=True, help="True if PO reception is confirmed by the vendor.")
+
+    receipt_reminder_email = fields.Boolean('Receipt Reminder Email', related='partner_id.receipt_reminder_email', related_sudo=True, readonly=False,
+        help='Automatically send a reminder email to your vendors before receipt')
+    reminder_date_before_receipt = fields.Integer('Days Before Receipt', related='partner_id.reminder_date_before_receipt', related_sudo=True, readonly=False,
+        help="Number of days to send reminder email before the promised receipt date")
+
+    def _send_reminder_mail(self):
+        template_name = 'purchase.email_template_edi_purchase_reminder'
+        try:
+            template = self.env.ref(template_name, raise_if_not_found=False)
+        except ValueError:
+            template = False
+
+        if template:
+            template.send_mail(self.id, raise_exception=False, force_send=True, notif_layout="mail.mail_notification_paynow")
+
+    @api.model
+    def send_reminder_mail(self):
+        if not self.user_has_groups('purchase.group_send_reminder') and not self.partner_id.receipt_reminder_email:
+            return
+
+        for order in self.get_orders_to_remind():
+            date = order.date_planned or order.expected_date
+            if (date - relativedelta(days=order.reminder_date_before_receipt)).date() == datetime.today().date():
+                order._send_reminder_mail()
+
+    @api.model
+    def get_orders_to_remind(self):
+        """When auto sending a reminder mail, only send for purchase order and
+        not all products are service."""
+        return self.search([('state', 'in', ['purchase', 'done'])]).filtered(lambda p: p.mapped('order_line.product_id.product_tmpl_id.type') != 'service')
+
+    @api.onchange('date_planned')
+    def onchange_date_planned(self):
+        if self.date_planned:
+            self.order_line.date_planned = self.date_planned
+
+    def get_confirm_url(self, confirm_type=None):
+        url = self.get_portal_url()
+        if confirm_type:
+            url += "&confirm=" + confirm_type
+        return url
+
+    def get_update_url(self):
+        return self.get_portal_url() + "&update=True"
```

Comment:
it would be safer to use werkzeug urls join + encode, but if you think it'd be overkill it's fine by me